networkit/cpp/graph/GraphTools.cpp

```diff
@@ -74,6 +77,39 @@ Graph restoreGraph(std::vector<node>& invertedIdMap, const Graph& G) {
 	return Goriginal;
 }
 
+Graph extractLargestConnectedComponent(const Graph &G, bool compactGraph) {
+	Graph S(G);
+
+	if (!G.numberOfNodes())
+		return S;
+
+	ConnectedComponents cc(S);
+	cc.run();
+
+	auto compSizes = cc.getComponentSizes();
+	auto largestCC = std::max_element(
+		compSizes.begin(), compSizes.end(),
+		[](const std::pair<index, count> &x, const std::pair<index, count> &y) {
+			return x.second < y.second;
+	});
+
+	count nodesToRemove = G.numberOfNodes() - largestCC->second;
+	S.forNodesWhile(
```

Comment:
I don't think it is supported to remove nodes while iterating over them. Instead, we should iterate over the component and call `removeNode`. This would also be faster.